pkgs/applications/science/physics/crystfel/default.nix

```
@@ -0,0 +1,133 @@
+    # We hard-code this by providing a little patch and then passing the absolute path to syminfo.lib as a
+    # preprocessor flag.
+    preBuild = ''
+      makeFlagsArray+=(CFLAGS='-DNIX_PROVIDED_SYMOP_FILE=\"$out/share/syminfo.lib\"')
+    '';
+
+    patches = [ ./libccp4-use-hardcoded-syminfo-lib.patch ];
+  };
+  # This is the statically-linked, pre-built binary of mosflm since compiling it from source turned out
+  # to be herculean task.
+  mosflm = stdenv.mkDerivation rec {
+    name = "mosflm";
+    version = "7.4.0";
+
+    src = fetchurl {
+      # It's version 7.4.0 but the version string in the URL is "ver740", hence replaceStrings
+      url = "https://www.mrc-lmb.cam.ac.uk/${name}/${name}/ver${builtins.replaceStrings ["."] [""] version}/pre-built/${name}-linux-64-noX11.zip";
+      sha256 = "1rqh3nprxfmnyihllw31nb8i3wfhybmsic6y7z6wn4rafyv3w4fk";
+    };
+
+    phases = [ "unpackPhase" "installPhase" ];
+
+    nativeBuildInputs = [ unzip makeWrapper ];
+
+    # The zip file you can download contains just one file and no directories, hence sourceRoot = "." since
+    # the unpack phase expects a single (sub)directory in the zip file by default.
+    sourceRoot = ".";
+
+    # mosflm statically links against its own libccp4, which as the syminfo.lib environment variable problem.
+    # Here, we circumvent it by creating a little wrapper script that calls mosflm after setting the SYMINFO variable.
+    installPhase = ''
+      mkdir -p $out/bin
+      cp mosflm-linux-64-noX11 $out/bin/
+      makeWrapper $out/bin/mosflm-linux-64-noX11 $out/bin/mosflm --set SYMINFO ${libccp4}/share/syminfo.lib --add-flags -n
+    '';
+  };
+
+  xgandalf = stdenv.mkDerivation rec {
+    name = "xgandalf";
+    version = "c15afa2381d5f87d4aefcc8181a15b4a6fd3a955";
+    src = fetchurl {
+      url = "https://gitlab.desy.de/thomas.white/${name}/-/archive/${version}/${name}-${version}.tar.gz";
+      sha256 = "11i1w57a3rpnb4x5y4n8d3iffn5m9w1zydl69syzljdk3aqg2pv8";
+    };
+
+    # - pkg-config to find eigen
+    # - meson/ninja is the build system
+    # - why it needs git, I don't know
+    nativeBuildInputs = [ meson git pkg-config ninja ];
+    buildInputs = [ eigen ];
+  };
+
+  pinkIndexer = stdenv.mkDerivation rec {
+    name = "pinkindexer";
+    version = "8a828788f8272a89d484b00afbd2500c2c1ff974";
+    src = fetchurl {
+      url = "https://gitlab.desy.de/thomas.white/${name}/-/archive/${version}/${name}-${version}.tar.gz";
+      sha256 = "1mkgf1xd91ay0z0632kzxm0z3wcxf0cayjvs6a3znds72dkhfsyh";
+    };
+
+    # - pkg-config to find eigen
+    # - meson/ninja is the build system
+    nativeBuildInputs = [ meson pkg-config ninja ];
+    buildInputs = [ eigen ];
+  };
+
+  fdip = stdenv.mkDerivation rec {
+    name = "fdip";
+    version = "4469a69d13eb529b52aa15e50847f302b1520014";
+    src = fetchurl {
+      url = "https://gitlab.desy.de/thomas.white/${name}/-/archive/${version}/${name}-${version}.tar.gz";
```

I wanted to visit the repo to figure out why it needs git but since you substituted nane I can't easily do that.

```suggestion
    pname = "fdip";
    version = "4469a69d13eb529b52aa15e50847f302b1520014";
    src = fetchurl {
      url = "https://gitlab.desy.de/thomas.white/fdip/-/archive/${version}/fdip-${version}.tar.gz";
```